client/src/services/getDishPairingApi.js

```diff
@@ -0,0 +1,24 @@
+import upsertDishPairing from './upsertDishPairing'
+
+export default function getDishPairingApi(wine_type) {
+  const wineTypeEncoded = wine_type.replace(' ', '_')
+  const { REACT_APP_API_SPOONACULAR_KEY2 } = process.env
+  const { REACT_APP_API_SPOONACULAR_BASEURL } = process.env
+
+  const FETCH_URL = `${REACT_APP_API_SPOONACULAR_BASEURL}/food/wine/dishes?apiKey=${REACT_APP_API_SPOONACULAR_KEY2}&wine=${wineTypeEncoded}`
+
+  return fetch(FETCH_URL)
+    .then(async res => {
+      if (res.ok) {
+        return await res.json()
+      } else {
+        const errorMessage = await res.text()
+        return Promise.reject(new Error(errorMessage))
+      }
+    })
+    .then(data => handleData(wineTypeEncoded, data))
```

Comment:
Why the mix of async/await and promises?
```suggestion
    .then(res => res.ok ? res.json() : throw new Error(res.text())
    .then(data => handleData(wineTypeEncoded, data))
```
💯  for rejecting with an error! Throwing an error inside `then` will reject the promise with that error.